tools/smoke_test.py

```diff
@@ -0,0 +1,166 @@
+import os
+import re
+import subprocess
+import sys
+import traceback
+
+try:
+    import packaging.version
+except ImportError:
+    from pkg_resources import packaging
+Version = packaging.version.Version
```

Comment:
Add a comment as to when / why this might fail please.

Reply:
`from pkg_resources import packaging` should be sufficient (that is, no try/except needed) since it should be bundled in the python environment.

Reply:
why the first try then?